### PR TITLE
feat(Role): add `flags`

### PR DESCRIPTION
**Pull request by N1ckPro**
**Please describe the changes this PR makes and why it should be merged:**

- https://github.com/discord/discord-api-docs/pull/6269
- https://github.com/discordjs/discord-api-types/pull/782

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -161,6 +161,7 @@ import {
   APIGuildOnboardingPrompt,
   APIGuildOnboardingPromptOption,
   GuildOnboardingPromptType,
```
Comment (jaw0r3k) [inline packages/discord.js/typings/index.d.ts:163]: You have to make an enum for that
as updating API types is forbidden

File: packages/discord.js/src/util/RoleFlagsBitField.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Data structure that makes it easy to interact with a {@link Role#flags} bitfield.
+ * @extends {BitField}
+ */
+class RoleFlagsBitField extends BitField {
+  /**
+   * Numeric role flags.
+   * @type {RoleFlags}
+   * @memberof RoleFlagsBitField
+   */
+  static Flags = RoleFlags;
```
Comment (jaw0r3k) [inline packages/discord.js/src/util/RoleFlagsBitField.js:16]: Well kts v13
so maybe look at other bit fields

Comment (N1ckPro) [reply packages/discord.js/src/util/RoleFlagsBitField.js:16]: This PR is for v14 tho 🤔

File: packages/discord.js/typings/index.d.ts [hunk 3]
```diff
@@ -161,6 +161,7 @@ import {
...
+  // RoleFlags,
```
Comment (Qjuh) [inline packages/discord.js/typings/index.d.ts:164]: Why did you comment this out?

File: packages/discord.js/typings/index.d.ts [hunk 4]
```diff
@@ -2545,6 +2547,13 @@ export class Role extends Base {
   public toString(): RoleMention;
 }
 
+export type RoleFlagsString = any; // keyof typeof RoleFlags;
```
Comment (Qjuh) [inline packages/discord.js/typings/index.d.ts:2550]: Same for this, should be actually the type, not `any`.

**Review by Qjuh — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 5]
```diff
@@ -2545,6 +2547,13 @@ export class Role extends Base {
   public toString(): RoleMention;
 }
 
+export type RoleFlagsString = keyof typeof RoleFlags;
+
+export class RoleFlagsBitField extends BitField<RoleFlagsString> {
+  public static Flags: Record<RoleFlagsString, number>;
```
Comment (jaw0r3k) [inline packages/discord.js/typings/index.d.ts:2553]:
```suggestion
  public static Flags: typeof RoleFlags;
  ```

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**